example/spec/distributionAPI/distributionSpec.js

```diff
@@ -76,51 +97,109 @@ describe('Distribution API', () => {
 
     beforeAll(async () => {
       fileUrl = getDistributionFileUrl({
-        bucket: config.bucket,
+        bucket: protectedBucketName,
         key: fileKey
       });
       fileChecksum = await generateChecksumFromStream(
         'cksum',
         fs.createReadStream(require.resolve(s3Data[0]))
       );
+      privateFileUrl = getDistributionFileUrl({
+        bucket: privateBucketName, key: fileKey
+      });
+      publicFileUrl = getDistributionFileUrl({
+        bucket: publicBucketName, key: fileKey
+      });
     });
 
     afterAll(async () => {
       await accessTokensModel.delete({ accessToken });
     });
 
-    it('redirects to Earthdata login for unauthorized requests', async () => {
-      const response = await got(
-        fileUrl,
-        { followRedirect: false }
-      );
-      const authorizeUrl = new URL(response.headers.location);
-      expect(authorizeUrl.origin).toEqual(process.env.EARTHDATA_BASE_URL);
-      expect(authorizeUrl.searchParams.get('state')).toEqual(`/${config.bucket}/${fileKey}`);
-      expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
-    });
+    describe('an unauthorized user', () => {
+      it('redirects to Earthdata login for unauthorized requests', async () => {
+        const response = await got(
+          fileUrl,
+          { followRedirect: false }
+        );
+        const authorizeUrl = new URL(response.headers.location);
+        expect(authorizeUrl.origin).toEqual(process.env.EARTHDATA_BASE_URL);
+        expect(authorizeUrl.searchParams.get('state')).toEqual(`/${protectedBucketName}/${fileKey}`);
+        expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
+      });
 
-    it('redirecting to Earthdata login for unauthorized requests to /s3credentials endpoint.', async () => {
-      const response = await got(
-        `${process.env.DISTRIBUTION_ENDPOINT}/s3credentials`,
-        { followRedirect: false }
-      );
-      const authorizeUrl = new URL(response.headers.location);
-      expect(authorizeUrl.origin).toEqual(process.env.EARTHDATA_BASE_URL);
-      expect(authorizeUrl.searchParams.get('state')).toEqual('/s3credentials');
-      expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
+      it('redirects to Earthdata login for requests on /s3credentials endpoint.', async () => {
+        const response = await got(
+          `${process.env.DISTRIBUTION_ENDPOINT}/s3credentials`,
+          { followRedirect: false }
+        );
+        const authorizeUrl = new URL(response.headers.location);
+        expect(authorizeUrl.origin).toEqual(process.env.EARTHDATA_BASE_URL);
+        expect(authorizeUrl.searchParams.get('state')).toEqual('/s3credentials');
+        expect(authorizeUrl.pathname).toEqual('/oauth/authorize');
+      });
+
+      it('downloads a public science file', async () => {
+        accessToken = randomId('accessToken');
+        const s3SignedUrl = await invokeLambdaForS3SignedUrl(publicFileUrl, accessToken);
```

Comment:
This could be a separate ticket, but it occurs to me that if we have this helper, could we use it for all the tests in this file? And remove the startup of a local Express server for the distribution API?

Reply:
Not sure, I'll have to look at that.

Reply:
Yes.  Server removed and invokeLambdaForS3SignedUrl renamed invokeApiDistributionLambda.